source/slang/slang-lower-to-ir.cpp

```diff
@@ -2573,6 +2611,74 @@ void addArg(
     }
 }
 
+void compareParamToArgumentMemoryQualifier(
```

Comment:
Doing type check at IR lowering time is usually too late. You should be able to check this during visitInvokeExpr.